### PR TITLE
Remove extern ter ids

### PR DESCRIPTION
**Pull request by Procyonae**
#### Summary
None

#### Purpose of change

Remove everything but t_null from the ancient (Pre 0.A) int_id<ter_t> list from mapdata.h/.cpp
Extern evil idk

#### Describe the solution

Changes references to the extern list for static const string_id<ter_t> definitions at the top of each file that uses them in line with other ids
Uses ter_str_id::NULL_id() instead for t_null

Left a few instances of t_null that were being awkward, if someone more technically adept wants to attack them here's the list

mapdata.cpp L64
mapgen.cpp L854 L4376 L5144 L7690
regional_settings.h L100 L283
submap.h L318

Changes a few excessive checks to use unordered_set and find instead

#### Describe alternatives you've considered

I were going to just remove the ones that aren't referenced anymore but anothersimulacrum and kevingranade both advocated for changing to using string_id<ter_t>s per file

Keeping the ids as t_ as opposed to the currently used ter_t_ for string ids and instead changing existing ter_t_ to t_ but that's probably at least just as intrusive and other types of static id use the existing format

Not causing myself merge conflicts

#### Testing

Game compiles and loads fine

I wouldn't be surprised if there's some random place the int_id -> string_id causes an issue that I haven't noticed and the tests don't catch but hopefully it's nothing major

#### Additional context

There's alot of stuff that can be replaced with flag/ter_furn_transforms instead of needing to use hardcoded ids, might attack some in future PRs

### CODE REVIEW

**Review by nornagon — APPROVED** [review]
You're a saint

File: src/mapgen.cpp [hunk 1]
```diff
@@ -151,6 +151,37 @@ static const oter_str_id oter_tower_lab_stairs( "tower_lab_stairs" );
...
+static const ter_str_id ter_t_card_science( "t_card_science" );
+static const ter_str_id ter_t_concrete_wall( "t_concrete_wall" );
+static const ter_str_id ter_t_cvdbody( "t_cvdbody" );
+static const ter_str_id ter_t_cvdmachine( "t_cvdmachine" );
+static const ter_str_id ter_t_dirt( "t_dirt" );
+static const ter_str_id ter_t_door_glass_frosted_c( "t_door_glass_frosted_c" );
+static const ter_str_id ter_t_door_metal_c( "t_door_metal_c" );
+static const ter_str_id ter_t_door_metal_locked( "t_door_metal_locked" );
+static const ter_str_id ter_t_floor( "t_floor" );
+static const ter_str_id ter_t_fungus( "t_fungus" );
```
Comment (kevingranade) [inline src/mapgen.cpp:164]:
```suggestion
```

File: src/mapgen_functions.cpp [hunk 2]
```diff
@@ -52,6 +52,43 @@ static const oter_str_id oter_river_west( "river_west" );
 static const oter_str_id oter_slimepit( "slimepit" );
 static const oter_str_id oter_slimepit_down( "slimepit_down" );
 
+static const ter_str_id ter_t_buffer( "t_buffer" );
```
Comment (kevingranade) [inline src/mapgen_functions.cpp:55]:
```suggestion
```

File: src/mapgen_functions.cpp [hunk 3]
```diff
@@ -52,6 +52,43 @@ static const oter_str_id oter_river_west( "river_west" );
...
+static const ter_str_id ter_t_railroad_tie( "t_railroad_tie" );
+static const ter_str_id ter_t_railroad_tie_d( "t_railroad_tie_d" );
+static const ter_str_id ter_t_railroad_tie_d1( "t_railroad_tie_d1" );
+static const ter_str_id ter_t_railroad_tie_d2( "t_railroad_tie_d2" );
+static const ter_str_id ter_t_railroad_tie_h( "t_railroad_tie_h" );
+static const ter_str_id ter_t_railroad_tie_v( "t_railroad_tie_v" );
+static const ter_str_id ter_t_railroad_track( "t_railroad_track" );
+static const ter_str_id ter_t_railroad_track_d( "t_railroad_track_d" );
+static const ter_str_id ter_t_railroad_track_d1( "t_railroad_track_d1" );
+static const ter_str_id ter_t_railroad_track_d2( "t_railroad_track_d2" );
```
Comment (kevingranade) [inline src/mapgen_functions.cpp:73]:
```suggestion
```

File: src/mapgen_functions.cpp [hunk 4]
```diff
@@ -52,6 +52,43 @@ static const oter_str_id oter_river_west( "river_west" );
...
+static const ter_str_id ter_t_lava( "t_lava" );
+static const ter_str_id ter_t_null( "t_null" );
+static const ter_str_id ter_t_open_air( "t_open_air" );
+static const ter_str_id ter_t_railroad_rubble( "t_railroad_rubble" );
+static const ter_str_id ter_t_railroad_tie( "t_railroad_tie" );
+static const ter_str_id ter_t_railroad_tie_d( "t_railroad_tie_d" );
+static const ter_str_id ter_t_railroad_tie_d1( "t_railroad_tie_d1" );
+static const ter_str_id ter_t_railroad_tie_d2( "t_railroad_tie_d2" );
+static const ter_str_id ter_t_railroad_tie_h( "t_railroad_tie_h" );
+static const ter_str_id ter_t_railroad_tie_v( "t_railroad_tie_v" );
```
Comment (kevingranade) [inline src/mapgen_functions.cpp:69]:
```suggestion
```

File: src/recipe.cpp [hunk 5]
```diff
@@ -46,6 +46,8 @@ static const itype_id itype_null( "null" );
 
 static const std::string flag_FULL_MAGAZINE( "FULL_MAGAZINE" );
 
+static const ter_str_id ter_t_dirt( "t_dirt" );
```
Comment (kevingranade) [inline src/recipe.cpp:49]:
```suggestion
```

**Comment by Procyonae** [comment]
I have the next batch of clang fixes staged just waiting on the tests to finish in case there's more issues

**Pull request merged**